quantities.serialization/text.json.test/SerializationTest.cs

```diff
@@ -0,0 +1,210 @@
+
+    [Fact]
+    public void ReadPrefixedMetric()
+    {
+        Double value = Math.PI;
+        Length expected = Length.Of(value).Si<Centi, Metre>();
+
+        expected.SupportsSerialization();
+    }
+
+    [Fact]
+    public void ReadImperial()
+    {
+        Double value = Math.E;
+        Length expected = Length.Of(value).Imperial<Mile>();
+
+        expected.SupportsSerialization();
+    }
+
+    [Fact]
+    public void Metric()
+    {
+        Double value = Math.PI;
+        Length length = Length.Of(value).Si<Metre>();
+        String actual = length.Serialize(options);
+        String expected = $$"""
+        {
+          "length": {
+            "value": {{value:R}},
+            "si": {
+              "unit": "m"
+            }
+          }
+        }
+        """;
+        Assert.Equal(expected, actual);
+    }
+    [Fact]
+    public void MetricWithPrefix()
+    {
+        Double value = Math.PI;
+        Length length = Length.Of(value).Si<Kilo, Metre>();
+        String actual = length.Serialize(options);
+        String expected = $$"""
+        {
+          "length": {
+            "value": {{value:R}},
+            "si": {
+              "prefix": "K",
+              "unit": "m"
+            }
+          }
+        }
+        """;
+        Assert.Equal(expected, actual);
+    }
+    [Fact]
+    public void Imperial()
+    {
+        Double value = Math.PI;
+        Length length = Length.Of(value).Imperial<Yard>();
+        String actual = length.Serialize(options);
+        String expected = $$"""
+        {
+          "length": {
+            "value": {{value:R}},
+            "imperial": {
+              "unit": "yd"
+            }
+          }
+        }
+        """;
+        Assert.Equal(expected, actual);
+    }
+
+    [Fact]
+    public void PowerRepresentationsSupported()
+    {
+        Volume volume = Volume.Of(2).Cubic.Si<Metre>();
+
+        Volume roundRobinSerialization = volume.SerializeRoundRobin();
+
+        Assert.Equal(volume.ToString(), roundRobinSerialization.ToString());
+    }
+
+    [Fact]
+    public void Compound()
+    {
+        Double value = Math.PI;
+        Velocity velocity = Velocity.Of(value).Si<Kilo, Metre>().Per.Metric<Hour>();
+        String actual = velocity.Serialize(options);
+        String expected = $$"""
+        {
+          "velocity": {
+            "value": {{value:R}},
+            "frac": {
+              "si": {
+                "prefix": "K",
+                "unit": "m"
+              },
+              "metric": {
+                "unit": "h"
+              }
+            }
+          }
+        }
+        """;
+        Assert.Equal(expected, actual);
+    }
+    [Fact]
+    public void Complex()
+    {
+        var person = new Person {
+            Name = "Foo Bar",
+            Height = Length.Of(1.67).Si<Metre>(),
+            Weight = Mass.Of(72).Si<Kilogram>()
+        };
+        String actual = person.Serialize(options);
+        String expected = """
```

Comment:
I think this shows best how complex types are serialized. What do you think of the formatting I chose?

PS: for now I chose to not bother too much about pascal case, camel case, lower or upper case. It's just all lowercase... What do you think about this?

Reply:
I see some advantages to having everything lowercase:

- it will work well on both case sensitive & case insensitive systems
- the serialization of a TUnit is concise, so uppercase letters would bring no benefit
- it's simpler.